rust-src/concordium_base/src/common/cbor/encoder.rs

```diff
@@ -151,3 +173,41 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_map_wrong_size() {
+        let mut bytes = Vec::new();
+        let mut encoder = Encoder::new(&mut bytes);
+        let map_encoder = encoder.encode_map(1).unwrap();
+        let err = map_encoder.end().unwrap_err().to_string();
+        assert!(err.contains("expected map size 1"), "err: {}", err);
+    }
+
+    #[test]
+    fn test_array_wrong_size() {
+        let mut bytes = Vec::new();
+        let mut encoder = Encoder::new(&mut bytes);
+        let array_encoder = encoder.encode_array(1).unwrap();
+        let err = array_encoder.end().unwrap_err().to_string();
+        assert!(err.contains("expected array length 1"), "err: {}", err);
+    }
+
+    /// Test that map entries are ordered according to the rules
+    /// in <https://www.rfc-editor.org/rfc/rfc8949.html#name-core-deterministic-encoding>
+    #[test]
+    fn test_map_entry_order() {
+        let mut bytes = Vec::new();
+        let mut encoder = Encoder::new(&mut bytes);
+        let mut map_encoder = encoder.encode_map(3).unwrap();
+        map_encoder.serialize_entry(&256, &0).unwrap();
+        map_encoder.serialize_entry(&(256 * 256), &0).unwrap();
+        map_encoder.serialize_entry(&1, &0).unwrap();
+        map_encoder.end().unwrap();
+
+        assert_eq!(hex::encode(&bytes), "a30100190100001a0001000000");
+    }
```

Comment:
I think it would be better if we tested with keys where the lexical ordering is different from the numeric ordering, as here they are the same, i.e. in both cases they will be `1 < 256 < 65536`. Also would avoid the `256*256` as it's just extra overhead when looking through the test to verify the ordering manually 😛 

And for good measure, I think it would make sense to have a test with string keys as well.